src/utils.ts

```diff
@@ -77,62 +89,82 @@ const createSessionRequest = (
   ),
 })
 
-const createSessionAccount = async ({
+const buildSessionAccount = async ({
+  useCacheAuthorisation,
+  accountSessionSignature,
+  sessionRequest,
   provider,
+  address,
+  dappKey,
+}: CreateSessionParams): Promise<Account> => {
+  const argentBackendService = new ArgentBackendService(
+    ec.starkCurve.getStarkKey(dappKey),
```

Comment:
maybe only pass the dappPubKey when actually signing, together with the signature